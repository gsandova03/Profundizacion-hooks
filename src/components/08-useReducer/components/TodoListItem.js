import React from 'react';
import PropTypes from 'prop-types';

const TodoListItem = ({ id, index, todo, handleComplete, handleDelete }) => {
  return (
    <li className="list-group-item">
      <p 
        className={ `${ todo.done && 'completed' }` } 
        onClick={ () => handleComplete( id ) }
      > 
        { index + 1 }. { todo.todo } 
      </p>
      <button 
        className="btn btn-danger" 
        onClick={ () => handleDelete( id ) }
      > Eliminar </button>
    </li>
  )
}

TodoListItem.propTypes = {
  id: PropTypes.number.isRequired,
  index: PropTypes.number.isRequired,
  todo: PropTypes.object.isRequired,
  handleComplete: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired
}

export default TodoListItem;
