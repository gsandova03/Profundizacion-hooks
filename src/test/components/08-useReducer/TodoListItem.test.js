import React from 'react'
import { shallow } from "enzyme"
import { demoTodo } from '../../fixtures/demoTodo'
import TodoListItem from '../../../components/08-useReducer/components/TodoListItem'


describe('Pruebas en el componente <TodoListItem/>', () => {
  
  //jest.fn() permite hacer referenica a las funciones que pasamos como argumento a nuestro componente
  const handleComplete = jest.fn()
  const handleDelete = jest.fn()

  let wrapper = shallow( 
    <TodoListItem
      id={ 1 }
      todo={ demoTodo[0] }
      index={ 0 }
      handleComplete={ handleComplete }
      handleDelete={ handleDelete }
    /> 
  )

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow( 
      <TodoListItem
        id={ 1 }
        todo={ demoTodo[0] }
        index={ 0 }
        handleComplete={ handleComplete }
        handleDelete={ handleDelete }
      /> 
    )
  })

  test('debe de renderizarce correctamente', () => {
    expect( wrapper ).toMatchSnapshot();
  })

  test('debe de llamar la funcion handleDelete', () => {
    wrapper.find('button').simulate('click')
    expect( handleDelete ).toHaveBeenCalledWith( demoTodo[0].id )
    expect( handleDelete ).toHaveBeenCalledTimes(1)
    
  })

  test('debe de llamar la funcion handleToggle', () => {
    wrapper.find('p').simulate('click')
    expect( handleComplete ).toHaveBeenCalledWith( demoTodo[0].id )
  })

  test('debe de mostrar el texto correctamente ', () => {
    const texto = wrapper.find('p')
    expect( texto.text().trim() ).toBe( `1. ${ demoTodo[0].todo }` )
  })

  test('debe de tener la clase "completed" si el todo done es == true ', () => {
    
    const todo = demoTodo[0]
    todo.done = true

    wrapper = shallow( 
      <TodoListItem
        id={ 1 }
        todo={ todo }
        index={ 0 }
        handleComplete={ handleComplete }
        handleDelete={ handleDelete }
      /> 
    )

    expect( wrapper.find('p').hasClass('completed') ).toBe( true )

  })
  
})
