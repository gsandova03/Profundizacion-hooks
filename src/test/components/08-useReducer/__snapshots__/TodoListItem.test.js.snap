// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <TodoListItem/> debe de renderizarce correctamente 1`] = `
<li
  className="list-group-item"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    aprender react
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
     Eliminar 
  </button>
</li>
`;
